libs/wingc/src/type_check.rs

```diff
@@ -2584,6 +2596,49 @@ impl<'a> TypeChecker<'a> {
 		}
 	}
 
+	/// Validate that the given type is a subtype (or same) as the expected type, disregarding whether
+	/// the object is mutable or immutable. If not, add an error to the diagnostics.
+	///
+	/// Returns the given type on success, otherwise returns the expected type.
+	fn validate_type_binary_operator(
+		&mut self,
+		actual_type: TypeRef,
+		expected_type: TypeRef,
+		span: &impl Spanned,
+	) -> TypeRef {
+		if actual_type.is_array_type()
+			&& expected_type.is_array_type()
+			&& actual_type
+				.collection_item_type()
+				.unwrap()
+				.is_same_type_as(&expected_type.collection_item_type().unwrap())
```

Comment:
The inner types don't need to be the same, you should be able to compare subtypes as well. So you should still use `self.validate_type_in` for the inner types of the actual and expected